Index: src/main/java/org/example/UIProcessor/ConsoleUI/ConsoleUI.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package org.example.UIProcessor.ConsoleUI;\r\n\r\npublic class ConsoleUI {\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/org/example/UIProcessor/ConsoleUI/ConsoleUI.java b/src/main/java/org/example/UIProcessor/ConsoleUI/ConsoleUI.java
--- a/src/main/java/org/example/UIProcessor/ConsoleUI/ConsoleUI.java	(revision ac7b248081c81f78bb98905c31c74eb0a9e3c40b)
+++ b/src/main/java/org/example/UIProcessor/ConsoleUI/ConsoleUI.java	(date 1732353965817)
@@ -1,4 +1,46 @@
 package org.example.UIProcessor.ConsoleUI;
 
-public class ConsoleUI {
+import org.example.UIProcessor.UIProcessor;
+
+import java.util.*;
+
+public class ConsoleUI implements UIProcessor {
+    Map<String,Runnable> options = new LinkedHashMap<>();
+
+    @Override
+    public void start() {
+
+    }
+
+    @Override
+    public void getInputFile() {
+
+    }
+
+    @Override
+    public void getOutputFile() {
+
+    }
+
+    @Override
+    public void stop() {
+
+    }
+
+    @Override
+    public void addOption(String name, Runnable runnable) {
+        options.put(name, runnable);
+    }
+
+    @Override
+    public void removeOption(String name) {
+
+    }
+
+    public void showOptions() {
+        System.out.println("Enter number of option \n" +
+                "Options:");
+        int i = 1;
+        options.forEach((key,value)->System.out.println(i+"."+key));
+    }
 }
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n<project xmlns=\"http://maven.apache.org/POM/4.0.0\"\r\n         xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n         xsi:schemaLocation=\"http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd\">\r\n    <modelVersion>4.0.0</modelVersion>\r\n\r\n    <groupId>org.example</groupId>\r\n    <artifactId>IndustialProject</artifactId>\r\n    <version>1.0-SNAPSHOT</version>\r\n\r\n    <properties>\r\n        <maven.compiler.source>23</maven.compiler.source>\r\n        <maven.compiler.target>23</maven.compiler.target>\r\n        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>\r\n    </properties>\r\n\r\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
--- a/pom.xml	(revision ac7b248081c81f78bb98905c31c74eb0a9e3c40b)
+++ b/pom.xml	(date 1732350219289)
@@ -14,4 +14,12 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     </properties>
 
+    <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.11.3</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
 </project>
\ No newline at end of file
Index: src/main/java/org/example/UIProcessor/UIProcessor.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package org.example.UIProcessor;\r\n\r\npublic interface UIProcessor {\r\n    void start();\r\n    void stop();\r\n    void addOption(String name, Runnable runnable);\r\n    void removeOption(String name);\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/org/example/UIProcessor/UIProcessor.java b/src/main/java/org/example/UIProcessor/UIProcessor.java
--- a/src/main/java/org/example/UIProcessor/UIProcessor.java	(revision ac7b248081c81f78bb98905c31c74eb0a9e3c40b)
+++ b/src/main/java/org/example/UIProcessor/UIProcessor.java	(date 1732353902122)
@@ -2,7 +2,7 @@
 
 public interface UIProcessor {
     void start();
+    void getInputFile();
+    void getOutputFile();
     void stop();
-    void addOption(String name, Runnable runnable);
-    void removeOption(String name);
 }
Index: src/test/java/org/example/UIProcessor/ConsoleUI/ConsoleUITest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package org.example.UIProcessor.ConsoleUI;\r\n\r\nclass ConsoleUITest {\r\n\r\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/org/example/UIProcessor/ConsoleUI/ConsoleUITest.java b/src/test/java/org/example/UIProcessor/ConsoleUI/ConsoleUITest.java
--- a/src/test/java/org/example/UIProcessor/ConsoleUI/ConsoleUITest.java	(revision ac7b248081c81f78bb98905c31c74eb0a9e3c40b)
+++ b/src/test/java/org/example/UIProcessor/ConsoleUI/ConsoleUITest.java	(date 1732351674538)
@@ -1,5 +1,13 @@
 package org.example.UIProcessor.ConsoleUI;
 
+import org.example.UIProcessor.UIProcessor;
+import org.junit.jupiter.api.Test;
+
 class ConsoleUITest {
+    UIProcessor consoleUI = new ConsoleUI();
 
+    @Test
+    void testAddOption() {
+        consoleUI.addOption("");
+    }
 }
\ No newline at end of file
